internal/bminventory/inventory.go

```diff
@@ -2444,6 +2410,10 @@ func (b *bareMetalInventory) updateNetworkParams(params installer.V2UpdateCluste
 			updates["api_vip"] = ""
 			updates["ingress_vip"] = ""
 			cluster.MachineNetworks = []*models.MachineNetwork{}
+			emptyCluster := common.Cluster{Cluster: models.Cluster{ID: cluster.ID}}
+			if err = network.UpdateVipsTables(db, &emptyCluster, true, true); err != nil {
+				return err
+			}
```

Comment:
I know the original comment `// VIP DHCP mode has changed` is not yours, but could you please expand it to something like

```
// VIP DHCP mode has changed. Because of this we are removing all the previously configured VIPs. This is because we do not want the non-DHCP mode to transparently use VIPs previously obtained via DHCP, nor we want DHCP mode to be fed with manually provided VIPs to start with
```

It is clear that this conditional fires when VIP DHCP mode has changed, but unless you know internals, it's difficult to unfold why the cleanup is happening. I hope putting a comment here can make it clear to whoever is touching this code in the future

Reply:
How about this wording?
> We are removing all the previously configured VIPs because the VIP
> DHCP mode has changed. Doing so will prevent situations where:
> 1. non-DHCP mode use VIPs previously obtained via DHCP transparently.
> 2. DHCP mode to be fed with manually provided VIPs.

Reply:
Done